src/main/scala/io/lambdaworks/workshop/exceptions/PasswordChecker.scala

```diff
@@ -2,15 +2,47 @@ package io.lambdaworks.workshop.exceptions
 
 object PasswordChecker {
 
-  def validate(password: String): Either[List[Throwable], String] = ???
-
-  private def minNumberOfChars(password: String, length: Int): Either[Throwable, String] = ???
-
-  private def containsUpperCase(password: String): Either[Throwable, String] = ???
-
-  private def containsLowerCase(password: String): Either[Throwable, String] = ???
-
-  private def containsNumber(password: String): Either[Throwable, String] = ???
+  def validate(password: String): Either[List[Throwable], String] = {
+    val passwordCheck1 = minNumberOfChars(password, 5)
+    val passwordCheck2 = containsUpperCase(password)
+    val passwordCheck3 = containsLowerCase(password)
+    val passwordCheck4 = containsNumber(password)
+    val checkerList    = List(passwordCheck1, passwordCheck2, passwordCheck3, passwordCheck4)
+    val errorList = checkerList
+      .collect({ case Left(x) => x })
+    if (errorList.nonEmpty) Left(errorList)
+    else Right(password)
+  }
+
+  private def minNumberOfChars(password: String, length: Int): Either[Throwable, String] =
+    if (password.length < length)
+      Left(InvalidLength)
+    else
+      Right(password)
+
+  private def containsUpperCase(password: String): Either[Throwable, String] = {
+    val hasUpperCase = password.exists(c => c.isUpper)
```

Comment:
You can omit variable here.
```suggestion
    val hasUpperCase = password.exists(_.isUpper)
```

Reply:
Same for other functions.